draft-ietf-lamps-x509-policy-graph.md

```diff
@@ -65,12 +65,25 @@ implementations vulnerable to denial-of-service attacks.
 
 # Introduction
 
-{{!RFC5280}} defines a suite of extensions for specifying certificate policies,
-along with a mechanism for mapping policies between subject and issuer policy
-domains in cross-certificates. This mechanism, when evaluated according to the
-algorithm in {{!RFC5280, Section 6.1}} produces a policy tree, describing
-policies asserted by each certificate, and mappings between them. This tree can
-grow exponentially in the depth of the certification path. This cost asymmetry
+{{!RFC5280}} defines a suite of extensions for determining the "policies" which
+apply to a certification path. A policy is described by an object identifier
+(OID), and a set of optional qualifiers.
+
+Policy validation in {{RFC5280}} is complex. As an overview, the certificate
+policies extension ({{Section 4.2.1.4 of !RFC5280}}) describes the policies,
+with optional qualifiers, under which an individual certificate was issued,
+and the policy mappings extension ({{Section 4.2.1.5 of !RFC5280}}) allows
+cross-certificates to map policy OIDs between subject and issuer policy
```

Comment:
```suggestion
certificates to map policy OIDs between subject and issuer policy
```

Not sure that the qualifier (word, not policy) is helping.

Reply:
I'm not sure where I got "cross-certificates" from instead of "CA certificates". Reading it without the qualifier, I think it may be misleading. But really the problem is the rest of sentence was _already_ misleading.

Suppose it said:

> ... allows certificates to map policy OIDs between subject and issuer policy domains

That reads, to me, as if the end-entity certificate is allowed to have this too. Suppose the CA has OID1, and the EE has OID2 but includes mappings (OID1 -> OID2). One might think, from reading that, that this checks out and we have an OID1 -> OID2 chain. But that would be off by one. The EE's OIDs match _first_. The mappings only apply to _subsequent_ certificates.

Because really the problem is that, although the fields are named subject and issuer, issuer actually refers to yourself and subject refers to the next one!

Perhaps:

> ... allows a CA certificate to map its policy OIDs to different policy OIDs in certificates that it issues

And then we also drop the confusing "policy domain" terminology which, IIRC, shows up in the ITU spec but not the IETF spec.